.github/workflows/build.yml

```diff
@@ -0,0 +1,83 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - "main"
+    paths-ignore:
+      - '.gitignore'
+      - 'LICENSE'
+      - '*.md'
+  pull_request:
+    paths-ignore:
+      - '.gitignore'
+      - 'LICENSE'
+      - '*.md'
+  workflow_dispatch:
+  schedule:
+    # Runs every Monday at 1:00 PM (GMT +08:00)
+    - cron: '0 5 * * 1'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+# In order to run tests against the real Azure services, you need to create a GitHub environment "ci" and add AZURE_CLIENT_ID / AZURE_TENANT_ID / AZURE_SUBSCRIPTION_ID as its variables to enable OIDC authentication with Azure.
+# See https://aka.ms/config-federated-identity-on-uami for more information on how to configure federated identity credential on a user-assigned managed identity (uami) and retrieve values for these required GitHub variables.
+# Besides, the uami needs to be granted the "Contributor" role on the subscription to deploy and delete Azure resources for testing.
+
+jobs:
+  build:
+    permissions:
+      id-token: write
+    environment: ci
+    env:
+      AZURE_CLIENT_ID: ${{ vars.AZURE_CLIENT_ID }}
+      AZURE_TENANT_ID: ${{ vars.AZURE_TENANT_ID }}
+      AZURE_SUBSCRIPTION_ID: ${{ vars.AZURE_SUBSCRIPTION_ID }}
```

Comment:
I request that you read these from secrets, as we have done in our other repositories, such as [liberty.aks](https://github.com/WASdev/azure.websphere-traditional.cluster/blob/f4103d4dd034174060b87a803efb5435355561bb/.github/workflows/integration-test.yaml#L64).

Reply:
I added them as variables as they are not secrets. I like your comment about the creation of setup and teardown scripts to set and clear these variables.

Reply:
Addressed in [a7fb09f](https://github.com/Azure/azure-microprofile/pull/7/commits/a7fb09fb209a2a94132cb75b0d8f348db56db7c4).